### PR TITLE
[stable/toxiproxy] - Adding new chart

### PR DESCRIPTION
**Pull request by nreymundo**
ToxiProxy is a proxy (duh) to simulate network issues and other situations. It can be use for chaos and load testing when wanting to mimic performance degradation or service outages. 

This is adding a new set of charts so that it can easily be deployed into a K8 environment. 

## Checklist

- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)
- [ ] Github actions are passing

### CODE REVIEW

File: stable/toxiproxy/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+name: toxiproxy
+description:  A TCP proxy to simulate network and system conditions for chaos and resiliency testing.
+version: "1.0"
+appVersion: "2.1.2"
```
Comment (max-rocket-internet) [inline stable/toxiproxy/Chart.yaml:5]: Can you add the extra fields like from here? https://github.com/deliveryhero/helm-charts/blob/master/stable/cachet/Chart.yaml

File: stable/toxiproxy/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+TODO
```
Comment (max-rocket-internet) [inline stable/toxiproxy/README.md:1]: This is autogenerated like this: `docker run --rm -v "$PWD:/helm-docs" jnorwood/helm-docs:latest --template-file=../../ci/README.md.gotmpl`

Details here: https://github.com/deliveryhero/helm-charts#contributing

File: stable/toxiproxy/templates/configmap-config.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "toxiproxy.fullname" . }}-config
```
Comment (max-rocket-internet) [inline stable/toxiproxy/templates/configmap-config.yaml:4]: Missing labels

File: stable/toxiproxy/templates/configmap-proxies.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "toxiproxy.fullname" . }}-proxies
```
Comment (max-rocket-internet) [inline stable/toxiproxy/templates/configmap-proxies.yaml:4]: Missing labels

File: stable/toxiproxy/templates/tests/test-connection.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
```
Comment (max-rocket-internet) [inline stable/toxiproxy/templates/tests/test-connection.yaml:1]: Let's delete this file, I don't think we need tests as it's covered in the CI of this repo.

File: stable/toxiproxy/values.yaml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+# to disable automatic downscaling change to true
+downscaler:
+  disabled: false
```
Comment (max-rocket-internet) [inline stable/toxiproxy/values.yaml:75]: kube-downscaler annotations would be passed in as values so perhaps add an option like `deploymentAnnotations: {}` in `values.yaml` and delete this:

```suggestion
deploymentAnnotations: {}
```

File: stable/toxiproxy/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,111 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "toxiproxy.fullname" . }}
+  labels:
+{{ include "toxiproxy.labels" . | indent 4 }}
+{{- if .Values.downscaler.disabled }}
+  annotations:
+    downscaler/exclude: "true"
+{{- end}}
```
Comment (max-rocket-internet) [inline stable/toxiproxy/templates/deployment.yaml:10]:
```suggestion
{{- with .Values.deploymentAnnotations }}
annotations:
  {{- toYaml . | nindent 4 }}
{{- end }}
```

**Review by max-rocket-internet — CHANGES_REQUESTED** [review]

File: stable/toxiproxy/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "toxiproxy.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "toxiproxy.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (nreymundo) [inline stable/toxiproxy/templates/deployment.yaml:21]: Getting a linting error here but not too sure what the problem is. 

```
[ERROR] templates/deployment.yaml: unable to parse YAML
	error converting YAML to JSON: yaml: line 21: mapping values are not allowed in this context

Error: 1 chart(s) linted, 1 chart(s) failed
```

File: stable/toxiproxy/templates/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,117 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "toxiproxy.fullname" . }}
+  labels:
+{{ include "toxiproxy.labels" . | indent 4 }}
+{{- with .Values.deploymentAnnotations }}
+annotations:
+  {{- toYaml . | nindent 4 }}
+{{- end }}
```
Comment (max-rocket-internet) [inline stable/toxiproxy/templates/deployment.yaml:10]:
```suggestion
{{- with .Values.deploymentAnnotations }}
  annotations:
  {{- toYaml . | nindent 4 }}
{{- end }}
```

File: stable/toxiproxy/templates/deployment.yaml (hunk 8 above)
Comment (max-rocket-internet) [reply stable/toxiproxy/templates/deployment.yaml:21]: Classic informative Helm error. I think my comment above.

Comment (nreymundo) [reply stable/toxiproxy/templates/deployment.yaml:21]: Fixed the wrong spacing on `annotations`. Still getting the same error though.

File: stable/toxiproxy/templates/deployment.yaml (hunk 9 above)
Comment (nreymundo) [reply stable/toxiproxy/templates/deployment.yaml]: Fixed I think.

**Review by max-rocket-internet — APPROVED** [review]
🚀

**Pull request merged**